src/simulation/routing/rust_road_update_router.rs

```diff
@@ -0,0 +1,166 @@
+use crate::simulation::messaging::events::EventsPublisher;
+use crate::simulation::messaging::messages::proto::simulation_update_message::Type;
+use crate::simulation::messaging::messages::proto::{TrafficInfoMessage, Vehicle};
+use crate::simulation::messaging::travel_time_collector::TravelTimeCollector;
+use crate::simulation::network::link::Link;
+use crate::simulation::network::network_partition::NetworkPartition;
+use crate::simulation::network::routing_kit_network::RoutingKitNetwork;
+use crate::simulation::routing::router::{CustomQueryResult, Router};
+use crate::simulation::routing::rust_road_router::RustRoadRouter;
+use crate::simulation::routing::traffic_message_broker::TrafficMessageBroker;
+use log::{debug};
+use mpi::topology::SystemCommunicator;
+use mpi::Rank;
+use std::collections::{HashMap, HashSet};
+use std::path::PathBuf;
+
+pub struct RustRoadUpdateRouter<'router> {
```

Comment:
I'm not sure whether this is a good name but no other came into my mind. Do you have one?